lms/djangoapps/certificates/api.py

```diff
@@ -880,7 +880,14 @@ def display_date_for_certificate(course, certificate):
 
     if _course_uses_available_date(course) and course.certificate_available_date < datetime.now(UTC):
         return course.certificate_available_date
-    elif course.certificates_display_behavior == CertificatesDisplayBehaviors.END and course.end:
+    # It is possible for a self-paced course run to end up configured with a display behavior of "END" even though it
+    # shouldn't be a valid option. We must check if the course is instructor-paced here to ensure that we are selecting
+    # the correct date to display.
+    elif (
+        not course.self_paced
+        and course.certificates_display_behavior == CertificatesDisplayBehaviors.END
+        and course.end
+    ):
```

Comment:
Fixing the issue with the data entering our system is beyond the scope of this PR. I remember I noticed this problem occurs when we create a new course run from an existing course. Some data created in Mongo uses default values and I believe this is generally how data like this gets created in the system.